Mathlib/Topology/Instances/CantorSet.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2024 Jana Göken. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artur Szafarczyk, Suraj Krishna M S, JB Stiegler, Isabelle Dubois,
+Tomáš Jakl, Lorenzo Zanichelli, Alina Yan, Emilie Uthaiwat, Jana Göken
+under guidance of Filippo A. E. Nuccio
+-/
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Algebra.GroupWithZero
+import Mathlib.Topology.Instances.Real
+
+/-!
+# Ternary Cantor Set
+This file contains the definition of the Cantor ternary set
+as well as some first properties, which will be updated later.
+
+## Main Definitions
+We give a definition for the ternary Cantor set in this file, which
+is `cantorSet`.
+-/
+
+/-- We define the preCantorSet as the preimages under the iterations of some functions. -/
+def preCantorSet : ℕ → Set ℝ
+  | 0 => Set.Icc 0 1
+  | Nat.succ n => (· / 3) '' preCantorSet n ∪ (fun x ↦ (2 + x) / 3)  '' preCantorSet n
+
+/-- We define the Cantor set as the limit of all preCantorSets. -/
+def cantorSet := iInf preCantorSet
+
+
+
+/-!
+## Simple Properties
+-/
+
+/-- The ternary Cantor set inherits the metric and in particular the topology from the reals. -/
+instance cantorSet.metricSpace : MetricSpace cantorSet := Subtype.metricSpace
+
+
+lemma quarters_mem_preCantorSet (n : ℕ) : 1/4 ∈ preCantorSet n ∧ 3/4 ∈ preCantorSet n := by
+  induction n with
+  | zero =>
```

Comment:
I'm sorry but I don't understand. ``preCantorSet 0`` is defined as ``Set.Icc 0 1``. What is the lemma that you think should be added? Something like "for every x in preCantorSet 0, x is greater than 0 and smaller than 1"? That's what's needed here.

Reply:
No no, I literally mean this lemma, is proof will indeed be `rfl`, since as you say it is true by definition. There are several reason to do that, for example tagging it `simp` will make possible to prove stuff like `0 ∈ preCantorSet 0` simply writing `simp` (now you have to write `simp [preCantorSet]`, that is not very longer, but things can be more complicated).